controllers/snapshotenvironmentbinding_controller.go

```diff
@@ -183,6 +183,23 @@ func detectTargetFromEnvironment(ctx context.Context, environment appstudiov1alp
 	}
 }
 
+func remoteSecretApplicableForEnvironment(remoteSecret rapi.RemoteSecret, environmentName string) bool {
+	var targetEnvironments []string
+	if environmentLabelInSecret, labelSet := remoteSecret.Labels[EnvironmentLabelName]; labelSet {
+		targetEnvironments = []string{environmentLabelInSecret}
+	} else if environmentAnnotationsInSecret, annotationSet := remoteSecret.Annotations[EnvironmentLabelName]; annotationSet {
+		envs := strings.Split(environmentAnnotationsInSecret, ",")
+		for i := range envs {
+			envs[i] = strings.TrimSpace(envs[i])
```

Comment:
I understand the idea behind allowing the whitespace, but I think we should be consistent in permitting (or forbidding) whitespace in comma-separated label and annotation values. And afaik, we mostly forbid it (e.g. in partial upload secrets) as we use the `commaseparated` package from RemoteSecret.